### PR TITLE
[bugfix] temporal columns with expression fail

### PR DESCRIPTION
**Pull request by mistercrunch**
error msg: "local variable 'literal' referenced before assignment"

Error occurs [only] when using temporal column defined as a SQL
expression.

Also noticed that examples were using `granularity` instead of using
`granularity_sqla` as they should. Fixed that here.

### CODE REVIEW

File: tests/model_tests.py [hunk 1]
```diff
@@ -105,3 +105,9 @@ def test_grains_dict(self):
         self.assertEquals(d.get('day').function, 'DATE({col})')
         self.assertEquals(d.get('P1D').function, 'DATE({col})')
         self.assertEquals(d.get('Time Column').function, '{col}')
+
+    def test_get_timestamp_expression(self):
+        tbl = self.get_table_by_name('birth_names')
+        ds_col = tbl.get_column('ds')
+        sqla_literal = ds_col.get_timestamp_expression(None)
```
Comment (hughhhh) [inline tests/model_tests.py]: Can we add a test thats passing a value into `get_timestamp_expression()` to make sure we are returning the proper `literal_col`

File: superset/connectors/sqla/models.py [hunk 2]
```diff
@@ -130,8 +131,9 @@ def get_timestamp_expression(self, time_grain):
                 elif pdf == 'epoch_ms':
                     expr = db_spec.epoch_ms_to_dttm().format(col=expr)
             grain = self.table.database.grains_dict().get(time_grain)
-            literal = grain.function if grain else '{col}'
-            literal = expr.format(col=expr)
+            if grain:
+                literal = grain.function
+        literal = expr.format(col=expr)
```
Comment (xrmx) [inline superset/connectors/sqla/models.py]: Isn't this line always overriding the one before?

File: tests/model_tests.py (hunk 1 above)
Comment (xrmx) [reply tests/model_tests.py]: Or just add one test per database:
https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures

File: superset/connectors/sqla/models.py (hunk 2 above)
Comment (betodealmeida) [reply superset/connectors/sqla/models.py]: Yeah, I think this should be

```python
literal = literal.format(col=expr)
```

Comment (betodealmeida) [reply superset/connectors/sqla/models.py]: See https://github.com/apache/incubator-superset/pull/4892

Comment (mistercrunch) [reply superset/connectors/sqla/models.py]: Yeah I think we're both wrong on this one :( , I'm going to clean this up a bit more and write many tests covering the whole function.

**Comment by mistercrunch** [comment]
superseeds https://github.com/apache/incubator-superset/pull/4892

**Review by betodealmeida — APPROVED** [review]

**Pull request merged**